onnxruntime/core/providers/coreml/builders/impl/conv_op_builder.cc

```diff
@@ -0,0 +1,154 @@
+
+  coreml_conv->set_isdeconvolution(false);
+
+  // Add Padding
+  // Usually using autopadding is more efficient than using explicit padding
+  // Try to see if we can map explicit padding to auto padding
+  std::vector<int64_t> input_shape;
+  ORT_RETURN_IF_ERROR(GetShape(*input_defs[0], input_shape));
+  AutoPadType auto_pad_type;
+  ORT_RETURN_IF_ERROR(HandleAutoPad(input_shape, weight_shape[2], weight_shape[3],
+                                    onnx_pads, strides, dilations,
+                                    StringToAutoPadType(helper.Get("auto_pad", "NOTSET")),
+                                    auto_pad_type));
+
+  if (AutoPadType::SAME_UPPER == auto_pad_type || AutoPadType::SAME_LOWER == auto_pad_type) {
+    auto* padding_type = coreml_conv->mutable_same();
+    if (AutoPadType::SAME_LOWER == auto_pad_type) {  // default is SAME_UPPER
+      padding_type->set_asymmetrymode(COREML_SPEC::SamePadding_SamePaddingMode_TOP_LEFT_HEAVY);
+    }
+  } else {
+    auto* padding_type = coreml_conv->mutable_valid();
+    if (AutoPadType::NOTSET == auto_pad_type && onnx_pads != std::vector<int64_t>{0, 0, 0, 0}) {
+      // NOTSET is adding the explicit padding to the ValidPadding.paddingAmounts
+      auto* heightBorder = padding_type->mutable_paddingamounts()->add_borderamounts();
+      heightBorder->set_startedgesize(onnx_pads[0]);
+      heightBorder->set_endedgesize(onnx_pads[2]);
+      auto* widthBorder = padding_type->mutable_paddingamounts()->add_borderamounts();
+      widthBorder->set_startedgesize(onnx_pads[1]);
+      widthBorder->set_endedgesize(onnx_pads[3]);
+    }
+  }
+
+  // Add weight
+  CreateCoreMLWeight(*coreml_conv->mutable_weights(), weight_tensor);
+
+  // Add bias if present
+  if (input_defs.size() > 2) {
+    coreml_conv->set_hasbias(true);
+    const auto& bias_tensor = *model_builder.GetInitializerTensors().at(input_defs[2]->Name());
+    CreateCoreMLWeight(*coreml_conv->mutable_bias(), bias_tensor);
+  }
+
+  *layer->mutable_input()->Add() = node.InputDefs()[0]->Name();
+  *layer->mutable_output()->Add() = node.OutputDefs()[0]->Name();
+
+  model_builder.AddLayer(layer.release());
+  return Status::OK();
+}
+
+// Operator support related
+
+bool ConvOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                                      const logging::Logger& logger) const {
+  const auto& name = node.Name();
+  const auto input_defs = node.InputDefs();
+
+  const auto& weight_name = input_defs[1]->Name();
+  if (Contains(initializers, weight_name)) {
+    const auto& tensor = *initializers.at(weight_name);
+    if (tensor.dims().size() != 4) {
+      LOGS(logger, VERBOSE) << "Conv [" << name << "] dimension: " << tensor.dims().size()
+                            << " Only conv 2d is supported.";
+      return false;
+    }
+  } else {
+    LOGS(logger, VERBOSE) << "The weight of Conv [" << name << "] must be known";
+    return false;
```

Comment:
May be clearer to say it must be a constant initializer or that it must be known prior to model execution